src/test/java/com/wikia/webdriver/common/core/Helios.java

```diff
@@ -206,9 +213,40 @@ private static String getUserIdUrl(String encodedUsername) {
       new UrlBuilder().getUrlForWiki("community"), encodedUsername, DateTime.now().getMillis());
   }
 
+  private static X509TrustManager getPassiveTrustManager() {
+    return new X509TrustManager() {
+
+      public X509Certificate[] getAcceptedIssuers() {
+        return null;
+      }
+
+      public void checkServerTrusted(X509Certificate[] arg0, String arg1)
+          throws CertificateException {
+      }
+
+      public void checkClientTrusted(X509Certificate[] arg0, String arg1)
+          throws CertificateException {
+      }
+    };
+  }
+
+  private static SSLContext createUnsecureSSLContext() {
+    SSLContext sc = null;
+    try {
+      TrustManager[] trustAllCerts = new TrustManager[]{Helios.getPassiveTrustManager()};
+      sc = SSLContext.getInstance("TLS");
+      sc.init(null, trustAllCerts, new SecureRandom());
+    } catch (Exception e) {
+      String msg = "error while creating unsecure SSLContext";
+      PageObjectLogging.log(msg, e, false);
+      throw new RuntimeException(msg, e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00112)